gensim/models/ldamodel.py

```diff
@@ -367,7 +367,15 @@ def inference(self, chunk, collect_sstats=False):
         # Lee&Seung trick which speeds things up by an order of magnitude, compared
         # to Blei's original LDA-C code, cool!).
         for d, doc in enumerate(chunk):
-            ids = [id for id, _ in doc]
+            if isinstance(doc, numpy.ndarray):
+                if len(doc) == 0:
+                    continue
+
+                # if utils.grouper was called with as_numpy=True all the
+                # term IDs will be floats, let's convert them to ints
+                ids = doc[:, 0].astype(numpy.int_)
```

Comment:
What if `doc` is only 1-dimensional? Also, probably better to cast to plain `int` instead of `numpy.int_`. What is `numpy.int_`?